File: problemas/iniciante/1049/1049.c
#include <string.h>
#include <stdio.h>

int main()
{
    char caracteristicas[3][20];

    for (int i = 0; i < 3; ++i)
    {
        scanf("%s\n", &caracteristicas[i]);
    }

    if (strcmp(caracteristicas[0], "vertebrado") == 0)
    {
        if (strcmp(caracteristicas[1], "ave") == 0)
        {
            if (strcmp(caracteristicas[2], "carnivoro") == 0)
            {
                printf("aguia\n");
            }
            else if (strcmp(caracteristicas[2], "onivoro") == 0)
            {
                printf("pomba\n");
            }
        }
        else if (strcmp(caracteristicas[1], "mamifero") == 0)
        {
            if (strcmp(caracteristicas[2], "onivoro") == 0)
            {
                printf("homem\n");
            }
            else if (strcmp(caracteristicas[2], "herbivoro") == 0)
            {
                printf("vaca\n");
            }
        }
    }
    else if (strcmp(caracteristicas[0], "invertebrado") == 0)
    {
        if (strcmp(caracteristicas[1], "inseto") == 0)
        {
            if (strcmp(caracteristicas[2], "hematofago") == 0)
            {
                printf("pulga\n");
            }
            else if (strcmp(caracteristicas[2], "herbivoro") == 0)
            {
                printf("lagarta\n");
            }
        }
        else if (strcmp(caracteristicas[1], "anelideo") == 0)
        {
            if (strcmp(caracteristicas[2], "hematofago") == 0)
            {
                printf("sanguessuga\n");
            }
            else if (strcmp(caracteristicas[2], "onivoro") == 0)
            {
                printf("minhoca\n");
            }
        }
    }

    return 0;
}

File: problemas/iniciante/1049/1049.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let arvore = {
    "vertebrado": {
        "ave": {
            "carnivoro": "aguia",
            "onivoro": "pomba"
        },
        "mamifero": {
            "onivoro": "homem",
            "herbivoro": "vaca"
        }
    },
    "invertebrado": {
        "inseto": {
            "hematofago": "pulga",
            "herbivoro": "lagarta"
        },
        "anelideo": {
            "hematofago": "sanguessuga",
            "onivoro": "minhoca"
        }
    }
};

let caracteristicas = [];

for (let i = 0; i < 3; ++i) {
    caracteristicas.push(lines.shift().trim());
}

console.log(arvore[caracteristicas[0]][caracteristicas[1]][caracteristicas[2]]);